src/LinAlg/SAM.C

```diff
@@ -957,19 +924,36 @@ Real SAM::normReact (const Vector& u, const Vector& rf) const
 #if SP_DEBUG > 1
         std::cout <<"SAM::normReact: idof="<< i+1 <<" SC="<< msc[i]
                   <<" u="<< u[i] <<" RF="<< rf(-msc[i]) <<" --> "
-                  << 0.5*retVal << std::endl;
+                  << retVal << std::endl;
 #endif
       }
 
-  return 0.5*retVal;
+  return retVal;
+}
+
+
+bool SAM::haveReaction (int dir, const IntVec* nodes) const
+{
+  if (dir > 0 && nspdof > 0)
+    for (int i = 0; i < nnod; i++)
+    {
+      int idof = madof[i]+dir-2;
```

Comment:
this looks suspicious, what happens if i == 0 and dir == 1 ?

Reply:
yes, it might look weird but I think it is right. It is because madof contains 1-based indices and dir is also a 1-based index whereas idof is to be 0-based. Therefore we subtract 2 and idof will never be less that 0 unless madof contains rubbish.